### PR TITLE
types(DataTable): Update Renderer data shape to allow generics.

### PR DESCRIPTION
**Pull request by milesj**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

Minor changes to some types so that the `rowData` property can be properly typed. This assumes the same shape for all renderers.

## Motivation and Context

Unknown is too restrictive.

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by schillerk** [comment]
Nice! Thanks for this

**Comment by Alphy11** [comment]
@Tgeo

File: packages/core/src/components/DataTable/story.tsx [hunk 1]
```diff
@@ -14,14 +14,27 @@ import Button from '../Button';
...
+  menu: string;
+  cats: number;
+};
+
+function CustomRenderer({ row, keyName }: RendererProps<CustomShape>) {
+  return <span>{String(row.rowData.data[keyName])}</span>;
+}
 
-const renderers = {
+const renderers: Renderers<CustomShape> = {
```
Comment (Alphy11) [inline packages/core/src/components/DataTable/story.tsx]: Should we default this?

**Comment by williaster** [comment]
nice

**Pull request merged**